server/service/src/sync/translations/period.rs

```diff
@@ -0,0 +1,97 @@
+use chrono::NaiveDate;
+use repository::{PeriodRow, StorageConnection, SyncBufferRow};
+use serde::{Deserialize, Serialize};
+
+use super::{
+    IntegrationRecords, LegacyTableName, PullDeleteRecordTable, PullUpsertRecord, SyncTranslation,
+};
+
+const LEGACY_TABLE_NAME: &'static str = LegacyTableName::PERIOD;
+
+fn match_pull_table(sync_record: &SyncBufferRow) -> bool {
+    sync_record.table_name == LEGACY_TABLE_NAME
+}
+
+#[allow(non_snake_case)]
+#[derive(Deserialize, Serialize)]
+pub struct LegacyPeriodRow {
+    #[serde(rename = "ID")]
+    pub id: String,
+    #[serde(rename = "periodScheduleID")]
+    pub period_schedule_id: String,
+    #[serde(rename = "startDate")]
+    pub start_date: NaiveDate,
+    #[serde(rename = "endDate")]
+    pub end_date: NaiveDate,
+    pub name: String,
+}
+
+pub(crate) struct PeriodTranslation {}
+impl SyncTranslation for PeriodTranslation {
+    fn try_translate_pull_upsert(
+        &self,
+        _: &StorageConnection,
+        sync_record: &SyncBufferRow,
+    ) -> Result<Option<IntegrationRecords>, anyhow::Error> {
+        if !match_pull_table(sync_record) {
+            return Ok(None);
+        }
+
+        let data = serde_json::from_str::<LegacyPeriodRow>(&sync_record.data)?;
+
+        let result = PeriodRow {
+            id: data.id.to_string(),
+            period_schedule_id: data.period_schedule_id.to_string(),
+            start_date: data.start_date,
+            end_date: data.end_date,
+            name: data.name.to_string(),
+        };
+
+        Ok(Some(IntegrationRecords::from_upsert(
+            PullUpsertRecord::Period(result),
+        )))
+    }
+
+    fn try_translate_pull_delete(
```

Comment:
period and period_schedules are not deleted in mSupply, they can only be deleted if they've been added before ok button is pressed